lib/services/role.go

```diff
@@ -1414,6 +1419,56 @@ func (set RoleSet) AdjustDisconnectExpiredCert(disconnect bool) bool {
 	return disconnect
 }
 
+// CheckDatabaseRoles returns whether a user should be auto-created in the
+// database and a list of database roles to assign.
+func (set RoleSet) CheckDatabaseRoles(database types.Database) (create bool, roles []string, err error) {
```

Comment:
should `database.GetAdminUser()` be checked here or somewhere? assuming a database that matches the role labels but has NO Admin user shouldn't create users.

Reply:
I'm not sure it belongs here tbh. It is checked as part of the user provisioning engine:

https://github.com/gravitational/teleport/blob/a1630b11f53c6460469e0669950ff1068c214918/lib/srv/db/postgres/users.go#L33-L35

Reply:
For example, say multiple Postgres databases match the same role and some have admin user set but others don't. Then the ones without admin user will always fail. Unless this is intended behaviour where all databases matching this role must have admin user set.

Reply:
Yeah, I think this would be expected behavior, right? Your role requires automatic user provisioning so if the database doesn't have an admin user configured it should fail.